lib/cdo/secrets.rb

```diff
@@ -144,12 +144,18 @@ def get_secret_value(client, key)
       raise
     end
 
-    # If +value+ is JSON, parse and wrap in ActiveSupport::OrderedOptions so
-    # property-method lookup chains are possible (e.g., secrets.secret.key).
+    # If +value+ is JSON, parse it and cast to a string.
+    #
+    # If +value+ is a JSON object (rather than a scalar), also wrap in
+    # ActiveSupport::OrderedOptions so property-method lookup chains are
+    # possible (e.g., secrets.secret.key).
+    #
     # @param value[String]
     # @return [ActiveSupport::OrderedOptions, String]
     def parse_json(value)
-      ActiveSupport::OrderedOptions[JSON.parse(value).symbolize_keys]
+      parsed = JSON.parse(value)
+      return parsed.to_s unless parsed.is_a?(Hash)
+      return ActiveSupport::OrderedOptions[parsed.symbolize_keys]
```

Comment:
Does this handle the scenario of the value being an array?

`['a', 2, 'c']`

Reply:
```
parsed = JSON.parse('["a", 2, "c"]')
puts parsed.is_a?(Hash) # false
puts parsed.is_a?(Array) # true
```

Reply:
Yes and no. It preserves the original behavior in that when given a string representing an array it will return a string representing an array rather than the array itself

Reply:
So we'd have to have special handling elsewhere. I'm guessing we don't have any actual examples of arrays in use?

Reply:
Correct. As I mentioned in the PR description, I don't know whether or not this helper method is actually desired; I just need to update the json gem

Reply:
Ah, no; what I said isn't quite right. The original behavior is to throw a `NoMethodError` when attempting to parse an array. Specifically:

   NoMethodError: undefined method `symbolize_keys' for [1, 2, 3]:Array

So, this does represent a change in behavior. I'm inclined given that to update this method to support arrays, since the existing functionality now seems obviously incomplete

Reply:
> inclined given that to update this method to support arrays

That's my vote, if it's not too deep of a rabbit hole.

Reply:
How much would break if we dropped the `to_s` and allowed this to return numbers and booleans too?

Reply:
I don't know; what I _do_ know is that we have some tests which explicitly check that numbers get cast to a string. See "Testing story" in the PR description for an example

Reply:
I do suspect that this was not _originally_ an intentional piece of functionality. Probably we just wanted something that worked for Strings or Hashes and didn't care about other data values.

Given how long this has been in place, though, it's quite possible that we have since come to rely on that behavior without realizing. In order to feel comfortable making the change you suggest, I'd have to be confident that we aren't doing that in any environment including production. And that sounds like more leg work than I want to do for a simple dependency update :smile: